test/testrecording.py

```diff
@@ -0,0 +1,26 @@
+class TestRecording:
+
+    def __init__(self, recordingId):
+        self.recordingId = recordingId
+
+    def is_tagged_as(self, animal):
+        return TestTagPromise(self, animal)
+        
+
+class TestTagPromise():
+    def __init__(self, testrecording, animal):
+        self._testrecording = testrecording
+        self._tagData = {"animal": animal}
+        print("  which is tagged as a {}".format(animal), end='')
+
+    def by(self, testuser):
+        print(" by {}".format(testuser.username))
+        testuser.tag_recording(self._testrecording.recordingId, self._tagData)
+        return self._testrecording
+
+    def byAI(self, testuser):
```

Comment:
It does this take `testuser`? It doesn't use it.

Maybe call this method `is_automatic()` or something?

Reply:
The code for doing this is all in userapi, so needs a user.   Without a user you have to somehow get the URL etc so I just took a short cut and get the caller to pass in the admin user.   I now it's not strictly necessary.   Feel free to change it.